### PR TITLE
G5 updates

### PR DESCRIPTION
**Pull request by johnbensnyder**
*Issue #, if available:*

*Description of changes:*

Current FSDP settings do not work on G5 instances. Added instructions for training on G5.

Also fixed issue with torchrun set to port 0.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by verdimrc — COMMENTED** [review]
From usability pov, I'd rather all these env vars taken out altogether, since with recent stacks, these env vars will be [automatically set](https://github.com/aws/aws-ofi-nccl/blob/master/doc/efa-env-var.md).

Even when accounting for those using older stacks, should we flip the messaging saying something along the line "if you use older stacks, additional env vars need to be set...".

I've seen other (reputable) repos keep using these env vars, which I guess is because they follow historical examples.

I'll let others to chip in, but my thinking is we can use this repo to advocate for new precedence for all the other users too.

File: 3.test_cases/10.FSDP/1.distributed-training.sbatch [hunk 1]
```diff
@@ -33,7 +34,7 @@ declare -a TORCHRUN_ARGS=(
     --nnodes=$SLURM_JOB_NUM_NODES \
     --rdzv_id=$SLURM_JOB_ID \
     --rdzv_backend=c10d \
-    --rdzv_endpoint=$(hostname):0 \
```
[the remark this answers is not in the document]
Comment (verdimrc) [reply 3.test_cases/10.FSDP/1.distributed-training.sbatch:36]: I don't see any issue with this, as long as nodes are allocated in exclusive mode.

**Review by verdimrc — APPROVED** [review]

File: 3.test_cases/10.FSDP/1.distributed-training.sbatch (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnbensnyder) [reply 3.test_cases/10.FSDP/1.distributed-training.sbatch:36]: I just tried running this again, but instead completely left the port off, and it seemed to work fine. I'll switch to that, and add a note in the readme about fixing the port number.

[the remark this answers is not in the document]
Comment (perifaws) [reply 3.test_cases/10.FSDP/1.distributed-training.sbatch]: Addressed in resolved confictl

**Pull request merged**